ktlint-rule-engine/src/main/kotlin/com/pinterest/ktlint/rule/engine/internal/KtlintSuppression.kt

```diff
@@ -202,12 +201,15 @@ private fun ASTNode.existingSuppressions() =
     existingSuppressionsFromNamedArgumentOrNull()
         ?: getValueArguments()
 
-private fun ASTNode.existingSuppressionsFromNamedArgumentOrNull() =
-    psi
-        .findDescendantOfType<KtCollectionLiteralExpression>()
-        ?.children
-        ?.map { it.text }
-        ?.toSet()
+private fun ASTNode.existingSuppressionsFromNamedArgumentOrNull(): Set<String>? =
+    recursiveChildren()
+        .firstOrNull { it.elementType == ElementType.COLLECTION_LITERAL_EXPRESSION }
+        ?.run {
+            children()
+                .filter { it.elementType == ElementType.STRING_TEMPLATE }
+                .map { it.text }
+                .toSet()
+        }
```

Comment:
Tnx for the Detekt reference. I now understand its usage better. It might take some time for me to get used too, but at first sight I find it harder to read. Lets keep it for now. If it bothers me in the future, I will remove it.